### PR TITLE
🔖 (release): v4.0.0rc1

### PR DESCRIPTION
**Pull request by schlich**

### CODE REVIEW

**Comment by dkirkham** [comment]
Hi @schlich, thanks for merging #472, but it wasn't _quite_ ready... as well as the docs, there are some packaging and testing changes. Many are in my local repo, so I'll now create separate PRs for those.

There are also some breaking changes, impacting anyone who has customised the admin, which would suggest we should be going for v4.0 rather than v3.2.0.

**Comment by schlich** [comment]
Thanks for the update @dkirkham -- the merge doesn't trigger a release or anything, and as far as I'm concerned as a maintainer if it passes CI it is safe to merge to main -- if you think the merge is in a broken state then let's make sure we capture those failure states in tests.

Thanks for the major release warning -- I had questions about whether this might be a minor or major release.  Can you point me to some code related to the breaking change? Let's make sure we have a migration process documented. 

Let me know if there's anything I can do to help with the packaging/testing changes you mention (maybe this resolves my comments above).

**Comment by MrCordeiro** [comment]
On that note, @schlich please allow at least one day between opening a PR and merging it so that it gets the chance  to be reviewed by other people.

This is what we've been following in this project for other PRs.

File: wagtailmenus/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # major.minor.patch.release.number
 # release must be one of alpha, beta, rc, or final
-VERSION = (3, 1, 9, "final", 0)
+VERSION = (3, 2, 0, "rc", 0)
```
Comment (MrCordeiro) [inline wagtailmenus/__init__.py:5]:
```suggestion
VERSION = (4, 0, 0, "rc", 0)
```
I think we can call this v4.0... By the way, don't forget to update the `CHANGELOG.md` and the documentation's release notes in this PR  😉

https://wagtailmenus.readthedocs.io/en/stable/releases/index.html

Comment (dkirkham) [reply wagtailmenus/__init__.py:5]: May I suggest the following CHANGELOG entry for my recent contribution:

```
* Fix [#459](https://github.com/jazzband/wagtailmenus/issues/459) by adding support for Wagtail 5.2, Wagtail 6.0 and Python 3.12.
```

Comment (MrCordeiro) [reply wagtailmenus/__init__.py:5]: I would just go with "Added support for x, y, z" and then add another entry saying 

```md
Fix [#459](https://github.com/jazzband/wagtailmenus/issues/459) by migrating to wagtail.snippets. Removed `wagtail_modeladmin` support.
```

That way it becomes more explicit.

**Comment by MrCordeiro** [comment]
@dkirkham, I think we are close to a release, right? :D

**Comment by dkirkham** [comment]
@MrCordeiro and @schlich, I'm happy with my changes, now that they are merged. I have retested with a full set of tests using `tox`, and then within my own project by installing the master branch with pip. All good.

As far as I can see, this PR needs to have:
- the version updated to 4.0rc (as per your suggestion)
- CHANGELOG updated with this change (changenote as above) and anything else in the release
- docs/sources/releases/4.0.rst with any other documentation changes also in this release

May I leave that to you or the other contributors?

**Comment by schlich** [comment]
Yep i can take care of that this evening, rock on

**Pull request merged**